streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1576,6 +1581,12 @@ void shutdown(final boolean clean) {
 
     private void shutdownStateUpdater() {
         if (stateUpdater != null) {
+            // If there are failed tasks handling them first
+            for (final StateUpdater.ExceptionAndTask exceptionAndTask : stateUpdater.drainExceptionsAndFailedTasks()) {
+                final Task failedTask = exceptionAndTask.task();
+                closeTaskDirty(failedTask, false);
+            }
+
```

Comment:
Below, there is `stateUpdater.shutdown(Duration.ofMillis(Long.MAX_VALUE))` -- should we set this to some smaller value, too?

Reply:
I think the MAX_VAUES were there in the beginning, and we never introduced a sane way to define a better timeout. Changing this now sounds like a good improvement